<center> <h1 style="background-color:#1C4D7E; padding: 4px; color:white" >Modélisation des pertes potentielles sur les prêts en banque</h1> 
    
<br>
  
<center><img
src="https://www.techfunnel.com/wp-content/uploads/2021/07/benefits_of_business_loans.png" 
             style="width:60%;height:60%;">
</center>
    
    
<div class="list-group" id="list-tab" role="tablist">
    
  <h3 class="list-group-item list-group-item-action active" style="background-color:#1C4D7E; color:white" data-toggle="list"  role="tab" aria-controls="home">Sommaire</h3> 
  <a class="list-group-item list-group-item-action" data-toggle="list" href="#introduction" role="tab" aria-controls="profile" style="color:#1C4D7E">Introduction<span class="badge badge-primary badge-pill" style="background-color:#1C4D7E; color:white">1</span></a>
  <a id="section2" class="list-group-item list-group-item-action" data-toggle="list" href="#question1" role="tab" aria-controls="messages" style="color:#333F4F">Question n°01<span class="badge badge-primary badge-pill" style="background-color:#1C4D7E; color:white">2</span></a>
  <a id="section2" class="list-group-item list-group-item-action" data-toggle="list" href="#question2" role="tab" aria-controls="messages" style="color:#1C4D7E">Question n°02<span class="badge badge-primary badge-pill" style="background-color:#1C4D7E; color:white">3</span></a>
  <a id="section2" class="list-group-item list-group-item-action" data-toggle="list" href="#question3" role="tab" aria-controls="messages" style="color:#1C4D7E">Question n°03<span class="badge badge-primary badge-pill" style="background-color:#1C4D7E; color:white">4</span></a>
  <a id="section2" class="list-group-item list-group-item-action" data-toggle="list" href="#question4" role="tab" aria-controls="messages" style="color:#1C4D7E">Question n°04<span class="badge badge-primary badge-pill" style="background-color:#1C4D7E; color:white">5</span></a>
    

<a id='introduction'></a>
# <div style=" background-color:#1C4D7E; color:white; padding: 12px; line-height: 1.5;">1. | Introduction</div>


## <div style=" background-color: #2F75BB; color:white; padding: 10px; line-height: 1.5;"> Énoncé
</div>
<div style="font-family: Segoe UI; line-height: 2; color: #000000; text-align: justify">
    
   👉 Une banque a accordé <span style="background-color: #C7DCF1; color: #000000;"><b>500 prêts</b></span> ce trimestre, et <span style="background-color: #C7DCF1; color: #000000;"><b>le taux de défaut sur ces prêts est estimé à 12%</b></span>. Si un prêt se solde par un défaut, la banque subira <span style="background-color: #C7DCF1; color: #000000;"><b>une perte de 30 000 €</b></span>. On suppose que chaque prêt est indépendant des autres et suit une distribution binaire (binomiale) avec <span style="background-color: #C7DCF1; color: #000000;"><b>une probabilité de défaut de 12%</b></span>.
</div>

## <div style=" background-color: #2F75BB; color:white; padding: 10px; line-height: 1.5;"> Importer les bibliothèques 📚
</div>

In [1]:
# Importer la fonction "comb" depuis le module math.

from math import comb

<div style="font-family: Segoe UI; line-height: 2; color: #000000; text-align: justify">
    
   <span style="background-color: #C7DCF1; color: #000000;"><b>"Comb"</b></span> est une fonction pour calculer <span style="background-color: #C7DCF1; color: #000000;"><b>le coefficient binomial.</b></span>
</div>

$$ C(n, k) = \frac{n!}{k!(n-k)!} $$


In [2]:
# Le nombre de combinaisons avec 5 tirages et avoir exactement 2 éléments.

comb(5, 2)

10

<div style="font-family: Segoe UI; line-height: 2; color: #000000; text-align: justify">
    
   Il y a 10 combinaisons possibles.
</div>

In [3]:
# La fonction de la loi binomiale :

def binomial_formula(n, k, p):
    return comb(n, k) * (p**k) * ((1-p)**(n-k)) 

$$ C(n, k) \cdot p^k \cdot (1-p)^{n-k} $$


<a id='question1'></a>
# <div style=" background-color:#1C4D7E; color:white; padding: 12px; line-height: 1.5;">2. | Question n°01</div>


<div style="font-family: Segoe UI; font-size: 16px; line-height: 2; color: #000000; text-align: justify">
Quelle est la probabilité qu'exactement <span style="background-color: #C7DCF1; color: #000000;"><b>60 prêts</b></span> parmi <span style="background-color: #C7DCF1; color: #000000;"><b>les 500 accordés se soldent par un défaut ?</b></span>
    </div>

In [10]:
round(binomial_formula(500, 60, 0.12),4)*100

5.48

<div style="font-family: Segoe UI; line-height: 2; color: #000000; text-align: justify">
    
   La probabilité qu’exactement 60 prêts se soldent par un défaut est de <span style="background-color: #C7DCF1; color: #000000;"><b>5.48%.</b></span>
</div>

In [11]:
# 60 prêts, chaque prêt est égale à une perte de 30 000€

60*30_000

1800000

<div style="font-family: Segoe UI; line-height: 2; color: #000000; text-align: justify">
    
   Ce qui correspond à une perte de <span style="background-color: #C7DCF1; color: #000000;"><b>1 800 000€.</b></span>
</div>

<a id='question2'></a>
# <div style=" background-color:#1C4D7E; color:white; padding: 12px; line-height: 1.5;">3. | Question n°02</div>


<div style="font-family: Segoe UI; font-size: 16px; line-height: 2; color: #000000; text-align: justify">
Quelle est la probabilité qu'au moins <span style="background-color: #C7DCF1; color: #000000;"><b>70 prêts</b></span> parmi <span style="background-color: #C7DCF1; color: #000000;"><b>les 500 accordés se soldent par un défaut ?</b></span>
    </div>

In [24]:
# Créer une liste vide qui va contenir toutes les probabilités supérieures ou égales à 70
plus_grand_que_70 = []

# Avec une boucle for, calculer chaque probabilité supérieur à 70
for i in range(70, 501):
    # Enregistrer chaque probabilité dans la liste
    plus_grand_que_70.append(binomial_formula(500, i, 0.12))

# Faire la somme de la liste
round(sum(plus_grand_que_70),5)*100
    

9.742

<div style="font-family: Segoe UI; line-height: 2; color: #000000; text-align: justify">
    
   La probabilité qu’au moins 70 prêts se soldent par un défaut est de <span style="background-color: #C7DCF1; color: #000000;"><b>9.74%.</b></span>
</div>

<a id='question3'></a>
# <div style=" background-color:#1C4D7E; color:white; padding: 12px; line-height: 1.5;">4. | Question n°03</div>


<div style="font-family: Segoe UI; font-size: 16px; line-height: 2; color: #000000; text-align: justify">
Quelle est la probabilité que la perte totale due aux défauts de prêt <span style="background-color: #C7DCF1; color: #000000;"><b>dépasse 1 000 000 € ?</b></span>
    </div>

In [30]:
# Le nombre de prêts qui égale à 1 000 000 € de perte est :
round(1_000_000/30_000, 2)

33.33

<div style="font-family: Segoe UI; line-height: 2; color: #000000; text-align: justify">
    
   Au bout de <span style="background-color: #C7DCF1; color: #000000;"><b>33.33 défauts,</b></span> la perte sera égale à 1 000 000 €.
</div>

In [33]:
# Créer une liste qui va contenir toutes les probabilités supérieures ou égales à 34
plus_grand_que_34 = []

# Avec une boucle for, calculer toutes les probabilités supérieures à 34
for i in range(34, 501):
    # Enregistrer chaque probabilité dans la liste
    plus_grand_que_34.append(binomial_formula(500, i, 0.12))

# Faire la somme de la liste
round(sum(plus_grand_que_34),6)*100

99.9956

<div style="font-family: Segoe UI; line-height: 2; color: #000000; text-align: justify">
    
   La probabilité que la perte totale due aux défauts de prêt dépasse 1 000 000 € est de <span style="background-color: #C7DCF1; color: #000000;"><b>99,99%.</b></span>
</div>

<a id='question4'></a>
# <div style=" background-color:#1C4D7E; color:white; padding: 12px; line-height: 1.5;">5. | Question n°04</div>


<div style="font-family: Segoe UI; font-size: 16px; line-height: 2; color: #000000; text-align: justify">
<u>Que pouvez-vous conclure sur la banque ?</u>
    </div>

<div style="font-family: Segoe UI; line-height: 2; color: #000000; text-align: justify">
    
    
La banque accorde <span style="background-color: #C7DCF1; color: #000000;"><b>trop de prêts qui se soldent par un défaut</b></span>. Le fait que la probabilité de perdre au moins 1 000 000 € est de 99%, nous fait réfléchir sur l’obtention des prêts accordées. Si vous êtes data scientist, vous devriez conseiller à la banque de <span style="background-color: #C7DCF1; color: #000000;"><b>créer un modèle de machine learning</b></span>, qui va détecter les clients qui sont plus susceptibles de faire défaut.
    <br>
<u>Cela aura 2 buts :</u> 
- <span style="background-color: #C7DCF1; color: #000000;"><b>Diminuer</b></span> la probabilité qu’un prêt se solde par un défaut.
- <span style="background-color: #C7DCF1; color: #000000;"><b>Mieux</b></span> gérer le risque d’un défaut, en appliquant un taux d’intérêt plus élevé.

    
Vous voyez comment un modèle de machine learning peut se créer. Nous avons un problème, <span style="background-color: #C7DCF1; color: #000000;"><b>des pertes conséquentes dues à une mauvaise gestion des prêts accordés</b></span>, le modèle de machine learning va détecter les clients les plus probables de faire défaut, ce qui sera un gain considérable pour l’entreprise.
</div>
